### PR TITLE
For free use

### PR DESCRIPTION
**Pull request by Rupreht**
- Fix add read pass from file use MYSQL_ENV_MYSQL_ROOT_PASSWORD_FILE
- fix permissions an archive from 600 -> 640 (use admin group)

### CODE REVIEW

**Comment by dsteinkopf** [comment]
Thank you very much for this contribution. I will have a deeper look at this at the weekend. To help me a bit: Could you give me a short explanation of why this fix is necessary?

File: loop.sh [hunk 1]
```diff
@@ -30,6 +30,10 @@ file_env 'MYSQL_CONNECTION_PARAMS'
 file_env 'MYSQL_HOST'
 file_env 'MYSQL_USER'
 
+if [ -f "$MYSQL_ENV_MYSQL_ROOT_PASSWORD_FILE" ]; then
```
Comment (thib3113) [inline loop.sh:33]: why ? it's already done by https://github.com/dsteinkopf/backup-all-mysql/pull/15/files#diff-35a96b9f0b8e5eca6ebf02f011573135a9075ac9c2a961932b6983bb408c1ab8L27 ?

**Comment by thib3113** [comment]
@dsteinkopf about the diff with `MYSQL_ENV_MYSQL_ROOT_PASSWORD_FILE`, I just check the code, and I think this is just a misspelling in the readme ? no ? ( it is `MYSQL_PASSWORD_FILE` ? ) . And finally the change in loop is not necessary, but needed in the readme ?

**Comment by dsteinkopf** [comment]
To be honest, I am not sure about this `MYSQL_ENV_MYSQL_ROOT_PASSWORD_FILE`. I just remember some Docker mechanism that automatically created and filled this kind of variables. I don't know if it is still useful for anyone to support this. As this script definitely is not very cleaned up or good code - but is still working well - I tend to just leave it and merge this PR.

**Comment by thib3113** [comment]
@dsteinkopf did you have more informations about this "mechanism" ?

Or you are speaking about the function file_env https://github.com/dsteinkopf/backup-all-mysql/blob/master/loop.sh#L9 , and when you call it the env name : https://github.com/dsteinkopf/backup-all-mysql/blob/master/loop.sh#L27 .

That will : read the content of the file `MYSQL_PASSWORD`_FILE` and set it in the env `MYSQL_PASSWORD` ?

**Comment by dsteinkopf** [comment]
> @dsteinkopf did you have more informations about this "mechanism" ?

No. But googleing for MYSQL_ENV_MYSQL_ROOT_PASSWORD gives some results...

> Or you are speaking about the function...

No. But thank you for your ideas and your help.

**Pull request merged**